modules/standard/Random.chpl

```diff
@@ -69,115 +68,152 @@ module Random {
 
-  // CHPLDOC FIXME: if the first line below is shifted one character to the
-  // left, it ends up being rendered like a method
+  private proc is1DRectangularDomain(d) param do
+    return d.isRectangular() && d.rank == 1;
 
-  // CHPLDOC FIXME: if the first line below is shifted one character to the
-  // right, it ends up causing a warning (promoted to error) in the .rst
-  // file.  It'd be preferable to have it declare the issue in terms of
-  // the .chpl line numbers.
-  //
 
-  private
-  proc isSupportedNumericType(type t) param do
-    return isNumericType(t) || isBoolType(t);
+  private proc oddTimeSeed(): int(64) {
+    use Time;
+    const seed = (timeSinceEpoch().totalSeconds()*1_000_000): int;
+    const oddseed = if seed % 2 == 0 then seed + 1 else seed;
+    return oddseed;
+  }
+
+  pragma "last resort"
+  @deprecated("The overload of `fillRandom` that accepts an 'algorithm' argument is deprecated; please remove the 'algorithm' argument")
+  proc fillRandom(ref arr: [], seed: int(64)=_SeedGenerator.oddCurrentTime, param algorithm=_defaultRNG)
+    where isNumericOrBoolType(arr.eltType)
+  {
+    var randNums = _createRandomStream(seed=seed,
+                                       eltType=arr.eltType,
+                                       parSafe=false,
+                                       algorithm=algorithm);
+    randNums.fillRandom(arr);
+  }
 
   /*
+    Fill a rectangular array of numeric values with pseudorandom values in
+    parallel using a new :record:`randomStream`. The first `arr.size` values
+    from the stream will be assigned to the array's elements in row-major
+    order. The parallelization strategy is determined by the array.
 
-    Fills a rectangular array of numeric elements with pseudorandom values in parallel using
-    a new stream implementing :class:`RandomStreamInterface` created
-    specifically for this call.  The first `arr.size` values from the stream
-    will be assigned to the array's elements in row-major order. The
-    parallelization strategy is determined by the array.
+    :arg arr: An array of numeric values
+    :arg seed: The seed to use when creating the ``randomStream``
 
-    .. note::
-      :mod:`NPBRandom` only supports `real(64)`, `imag(64)`, and `complex(128)`
-      numeric types. :mod:`PCGRandom` supports all primitive numeric types.
+  */
+  proc fillRandom(ref arr: [] ?t, seed: int)
+    where isNumericOrBoolType(t) && arr.isRectangular()
+  {
+    var rs = new randomStream(t, seed);
+    rs.fill(arr);
+  }
 
-    :arg arr: The array to be filled, where T is a primitive numeric type. Only
-      rectangular arrays are supported currently.
-    :type arr: `[] T`
+  /*
+    Fill a rectangular array of numeric values with pseudorandom values in
+    parallel using a new :record:`randomStream`. The first `arr.size` values
+    from the stream will be assigned to the array's elements in row-major
+    order. The parallelization strategy is determined by the array.
```

Comment:
IMO this needs to say something about the seed, like "the seed will be computed in an implementation-dependent manner"

Reply:
Agreed, I'll add those comments to all the places where a seed value is generated internally.